include/deal.II/dofs/dof_accessor.templates.h

```diff
@@ -312,111 +321,224 @@ namespace internal
             local_index,
           dof_handler.object_dof_indices[obj_level][d].size());
 
-        dof_handler.object_dof_indices
-          [obj_level][d]
-          [dof_handler.object_dof_ptr
-             [obj_level][d]
-             [dof_handler.hp_capability_enabled ?
-                (dof_handler.hp_object_fe_ptr[d][obj_index] + fe_index_) :
-                obj_index] +
-           local_index] = global_index;
+        process(
+          dof_handler.object_dof_indices
+            [obj_level][d]
+            [dof_handler.object_dof_ptr
+               [obj_level][d]
+               [dof_handler.hp_capability_enabled ?
+                  (dof_handler.hp_object_fe_ptr[d][obj_index] + fe_index_) :
+                  obj_index] +
+             local_index],
+          global_index);
       }
 
-
       /**
-       * Get the @p local_index-th degree of freedom corresponding to the
-       * finite element specified by @p fe_index on the vertex with global
-       * number @p vertex_index to @p global_index.
+       * Determine range of dofs of object in global data structure.
        */
       template <int dim, int spacedim, int d>
-      static types::global_dof_index
-      get_dof_index(const DoFHandler<dim, spacedim> &dof_handler,
-                    const unsigned int               obj_level,
-                    const unsigned int               obj_index,
-                    const unsigned int               fe_index,
-                    const unsigned int               local_index,
-                    const std::integral_constant<int, d> &)
+      DEAL_II_ALWAYS_INLINE static std::pair<unsigned int, unsigned int>
+      process_object_range(const DoFHandler<dim, spacedim> &dof_handler,
+                           const unsigned int               obj_level,
+                           const unsigned int               obj_index,
+                           const unsigned int               fe_index,
+                           const std::integral_constant<int, d> &)
```

Comment:
Can you explain why you need `always_inline` here? Did the compiler make bad decisions and are these functions performance-critical? I would have guess that at least for the simple cases (without the `find` or `distance`), the compiler should be able to apply early return and inline those parts, whereas the more expensive ones do not matter.

Reply:
I didn't have a look at the compiler output. My intention was that the early returns are inlined (because the return values there can be quite simply determined without an extra function call). But I can remove the `DEAL_II_ALWAYS_INLINE`.

Reply:
I guess that a reasonable optimizing compiler should inline the early returns; without measuring I would not use the `DEAL_II_ALWAYS_INLINE`. I say this as someone who has used it pretty excessively (after measuring).

Reply:
OK. I'll remove it. Are there anything else I should modify?

Reply:
@kronbichler  I have removed them!